crates/biome_analyze/src/lib.rs

```diff
@@ -825,12 +825,20 @@ pub struct AnalysisFilter<'a> {
     /// Only allow rules matching these names to emit signals
     pub enabled_rules: Option<&'a [RuleFilter<'a>]>,
     /// Do not allow rules matching these names to emit signals
-    pub disabled_rules: Option<&'a [RuleFilter<'a>]>,
+    pub disabled_rules: &'a [RuleFilter<'a>],
```

Comment:
Why did we change the type of `disabled_rules` and not the type of `enabled_rules`? It feels weird that they have two different types. If there's a reason, we should document it.

Reply:
Because it is are fundamentally different from each other. If `disabled_rules` is empty, this means that we disable no rules. Conversely, if  `enabled_rules` is empty, this means that no rule are enabled. In this later case, we need an `Option` to signal the fact that all rules are enabled.

Reply:
Thank you for the explanation. I suppose we should change the docs to explain this part.